lib_eio/path.ml

```diff
@@ -35,6 +35,27 @@ let mkdir ~perm ((t:#Fs.dir), path) =
     let bt = Printexc.get_raw_backtrace () in
     Exn.reraise_with_context ex bt "creating directory %a" pp (t, path)
 
+let mkdirs ?(exists_ok=false) ~perm ((t:#Fs.dir), path) =
+  let separate = String.split_on_char (Filename.dir_sep.[0]) in
+  let make_directory p =
+    try (
+      if exists_ok then
+        try t#mkdir ~perm p with Exn.Io ((Fs.E Fs.Already_exists _), _) -> ()
+      else
+        t#mkdir ~perm p
+    ) with Exn.Io _ as ex ->
+      let bt = Printexc.get_raw_backtrace () in
+      Exn.reraise_with_context ex bt "creating directory %a" pp (t, p)
+  in
+  let rec loop acc = function
+    | [] -> ()
+    | x :: xs ->
+      let dir = Filename.concat acc x in
+      make_directory dir;
+      loop dir xs
+  in
+  loop "" (separate path)
```

Comment:
It's probably more efficient to start by checking the direct parent. If that exists (which is the common) case, there's no need to check further up (see e.g. [Python's makedirs](https://github.com/python/cpython/blob/87de2fbb1e9b94f25ae8c0c233fb4690a13e4b02/Lib/os.py#L200) or [Bos's create](https://github.com/dbuenzli/bos/blob/6320f302eb9487518491511292a11ee6f04bd6e9/src/bos_os_dir.ml#L31)).